tools/sctool/src/main/java/com/horizen/ScBootstrappingToolCommandProcessor.java

```diff
@@ -393,27 +402,40 @@ private void processGenerateCertWithKeyRotationProofInfo(JsonNode json) throws E
                 return;
             }
 
-            mastersPublicKeys.add(pkNode.asText());
+            String masterKey = pkNode.asText();
+
+            int duplicateIndex = mastersPublicKeys.indexOf(masterKey);
+            if (duplicateIndex != -1) {
+                printGenerateCertWithKeyRotationProofInfoUsageMsg(String.format("duplicated keys in master keys are found. MasterKey with index %d equals to masterKey with index %d", duplicateIndex + 1, index));
+                return;
+            }
+
+            mastersPublicKeys.add(masterKey);
+            index++;
         }
 
-        assert mastersPublicKeys.size() == signersPublicKeys.size() : "mastersPublicKeys and signersPublicKeys must have the same size";
+        if (mastersPublicKeys.size() != signersPublicKeys.size()) {
+            printGenerateCertWithKeyRotationProofInfoUsageMsg(String.format("an amount of the signer keys and the master keys must be equal."));
+            return;
+        }
 
         for(int i = 0; i < mastersPublicKeys.size(); i++) {
-            if(Objects.equals(mastersPublicKeys.get(i), signersPublicKeys.get(i))) {
-                printGenerateCertWithKeyRotationProofInfoUsageMsg(String.format("signersKey with index %d equals to mastersKey with index %d", i, i));
+            int duplicateIndex = signersPublicKeys.indexOf(mastersPublicKeys.get(i));
+            if(duplicateIndex != -1) {
+                printGenerateCertWithKeyRotationProofInfoUsageMsg(String.format("duplicated keys are found. SignersKey with index %d equals to mastersKey with index %d", duplicateIndex, i));
                 return;
             }
         }
 
         if (!json.has("threshold") || !json.get("threshold").isInt()) {
-            printGenerateCertWithKeyRotationProofInfoUsageMsg("wrong threshold");
+            printGenerateCertWithKeyRotationProofInfoUsageMsg("threshold is missing or it has unsupported format");
             return;
         }
 
         int threshold = json.get("threshold").asInt();
 
         if (threshold <= 0 || threshold > signersPublicKeys.size()) {
-            printGenerateCertWithKeyRotationProofInfoUsageMsg("wrong threshold: " + threshold);
+            printGenerateCertWithKeyRotationProofInfoUsageMsg("threshold parameter should be greater than 0 and be less or equal to keyCount. Current value: " + threshold);
```

Comment:
"less or equal" change to "less than or equal"

Reply:
Done.